### PR TITLE
feat(RHOAIENG-21045): Add fairness metrics and tests

### LINKED ISSUE #30: Add fairness metrics and tests

### PR DESCRIPTION
**Pull request by christinaexyou**
## Summary by Sourcery

Implement a suite of group and individual fairness metrics with corresponding utility functions, and validate them through tests against AIF360 references.

New Features:
- Add group fairness metric classes: DisparateImpactRatio, GroupStatisticalParityDifference, GroupAverageOddsDifference, and GroupAveragePredictiveValueDifference
- Add IndividualConsistency class for measuring individual fairness

Enhancements:
- Introduce fairness_metrics_utils with row filtering and confusion matrix helpers

Tests:
- Add end-to-end tests comparing new fairness metrics to AIF360 implementations on a real dataset

### CODE REVIEW

File: src/core/metrics/fairness/fairness_metrics_utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import numpy as np
+
+def filter_rows_by_inputs(data, filter_func):
+    return data[np.apply_along_axis(filter_func, 1, data)]
+
+def calculate_confusion_matrix(test: np.array, truth: np.array, positive_class: int) -> dict:
```
Comment (RobGeada) [inline src/core/metrics/fairness/fairness_metrics_utils.py:7]: Maybe https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html instead?

File: src/core/metrics/fairness/group/disparate_impact_ratio.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import numpy as np
+
+class DisparateImpactRatio:
+    """
+    Calculate disparate impact ratio (DIR).
+    """
+    @staticmethod
+    def calculate_model(
+        samples: np.ndarray,
+        model: Any,
```
Comment (RobGeada) [inline src/core/metrics/fairness/group/disparate_impact_ratio.py:13]: Do we have a structure/planned datatype for this yet? While I like this idea, I fear it could run into the same security issues as we've been seeing with the explainers in the Java service

File: tests/metrics/data/data.csv
Comment (RobGeada) [inline tests/metrics/data/data.csv:1]: Are these to test parity against the existing Java service? I wonder if we might want to use synthetically generated data to avoid storing a lot of raw datasets in the repo

**Review by RobGeada — CHANGES_REQUESTED** [review]

**Comment by ruivieira** [comment]
@sourcery-ai review

**Review by ruivieira — APPROVED** [review]
@christinaexyou Let's merge as it is and address any issues on separate PRs 👍

**Review by RobGeada — APPROVED** [review]
